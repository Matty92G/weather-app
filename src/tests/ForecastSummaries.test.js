import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = [
    {
      date: 11111111,
      icon: "stubIcon1",
      description: "stubDescription1",
      temperature: {
        min: 10,
        max: 91,
      },
    },
    {
      date: 22222222,
      icon: "stubIcon2",
      description: "stubDescription2",
      temperature: {
        min: 20,
        max: 92,
      },
    },
  ];
  it("renders and matches snapshot", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
