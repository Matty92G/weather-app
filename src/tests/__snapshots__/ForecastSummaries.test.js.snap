// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        11111111
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        stubIcon1
      </div>
      <div
        class="forecast-summary_temperature"
      >
        91°C
      </div>
      <div
        class="forecast-summary_description"
      >
        stubDescription1
      </div>
    </div>
    <div
      class="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        22222222
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        stubIcon2
      </div>
      <div
        class="forecast-summary_temperature"
      >
        92°C
      </div>
      <div
        class="forecast-summary_description"
      >
        stubDescription2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
